fastlane/lanes/release.rb

```diff
@@ -278,6 +278,17 @@
     # Parse the provided version into an AppVersion object
     parsed_version = VERSION_FORMATTER.parse(new_version)
     previous_version = VERSION_FORMATTER.release_version(VERSION_CALCULATOR.previous_patch_version(version: parsed_version))
+    previous_release_branch = compute_release_branch_name(options: options, version: previous_version)
+
+    # Determine the base for the hotfix branch: either a tag or a release branch
+    base_ref_for_hotfix = if git_tag_exists(tag: previous_version)
+                            previous_version
+                          elsif Fastlane::Helper::GitHelper.branch_exists_on_remote?(branch_name: previous_release_branch)
+                            UI.message("ℹ️  Tag #{previous_version} not found. Using release branch #{previous_release_branch} as the base for hotfix instead.")
+                            previous_release_branch
+                          else
+                            UI.user_error!("Neither tag #{previous_version} nor branch #{previous_release_branch} exists! A hotfix branch cannot be created.")
+                          end
```

Comment:
[Testing this locally](https://github.com/wordpress-mobile/WordPress-iOS/pull/24996/#pullrequestreview-3475123846), I realized the difference in behavior base on whether the tag or branch exists locally only or on the remote, too. I think this is an important distinction to make. What do you think of adding it to the message?

```suggestion
                          elsif Fastlane::Helper::GitHelper.branch_exists_on_remote?(branch_name: previous_release_branch)
                            UI.message("ℹ️  Tag '#{previous_version}' not found on the remote. Using release branch '#{previous_release_branch}' as the base for hotfix instead.")
                            previous_release_branch
                          else
                            UI.user_error!("Neither tag '#{previous_version}' nor branch '#{previous_release_branch}' exists on the remote! A hotfix branch cannot be created.")
                          end
```

Reply:
Good point. Also with b0ee000bf125549696cc1712d0a8be02b878e34a we're specifically checking on remote, to make things clear.